### PR TITLE
support auto complete with facets

### PR DESCRIPTION
**Pull request by camelliazhang**
allow read facet name in autocomplete/datasets api

### CODE REVIEW

File: wherehows-dao/src/main/java/wherehows/dao/table/SearchDao.java [hunk 1]
```diff
@@ -34,20 +34,20 @@
...
+  public List<String> getAutoCompleteListDataset(String elasticSearchUrl, String input, int limit, String facet) {
+    return getAutoCompleteListbyES(elasticSearchUrl, facet, input, limit);
   }
 
   public List<String> getAutoCompleteListbyES(String elasticSearchUrl, String fieldName, String input, int limit) {
     // use elastic search completion suggester, ES will validate the input and limit
     List<String> completionSuggestionList = new ArrayList<>();
     Set<String> completionSuggestionSet = new HashSet<>();
 
+    fieldName = "name_suggest";
```
Comment (jywadhwani) [inline wherehows-dao/src/main/java/wherehows/dao/table/SearchDao.java]: Why are we setting the field name when it's already passed as an input?

Comment (camelliazhang) [reply wherehows-dao/src/main/java/wherehows/dao/table/SearchDao.java]: in os searchdao, the value is overwritten to be backward compatible as it only uses name_suggest.

File: wherehows-dao/src/main/java/wherehows/dao/table/SearchDao.java [hunk 2]
```diff
@@ -34,20 +34,20 @@
...
-  public List<String> getAutoCompleteList(String elasticSearchUrl, String input, int limit) {
-    return getAutoCompleteListDataset(elasticSearchUrl, input, limit / 3);
+  public List<String> getAutoCompleteList(String elasticSearchUrl, String input, int limit, String facet) {
+    return getAutoCompleteListDataset(elasticSearchUrl, input, limit / 3, facet);
   }
 
-  public List<String> getAutoCompleteListDataset(String elasticSearchUrl, String input, int limit) {
-    String fieldName = "name_suggest";
-    return getAutoCompleteListbyES(elasticSearchUrl, fieldName, input, limit);
+  public List<String> getAutoCompleteListDataset(String elasticSearchUrl, String input, int limit, String facet) {
```
[the remark this answers is not in the document]
Comment (camelliazhang) [reply wherehows-dao/src/main/java/wherehows/dao/table/SearchDao.java]: We could, curious that does the parameters need to be ordered based on the sequence of usage?

[the remark this answers is not in the document]
Comment (camelliazhang) [reply wherehows-dao/src/main/java/wherehows/dao/table/SearchDao.java]: Good point! Love to take it

**Pull request merged**